controllers/components/kserve/kserve_support.go

```diff
@@ -0,0 +1,280 @@
+package kserve
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/go-multierror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/manifest"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/serverless"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/resources"
+)
+
+var serviceAccounts = map[cluster.Platform][]string{
+	cluster.Unknown:          {odhModelControllerComponentName},
+	cluster.OpenDataHub:      {odhModelControllerComponentName},
+	cluster.ManagedRhods:     {odhModelControllerComponentName},
+	cluster.SelfManagedRhods: {odhModelControllerComponentName},
+}
+
+var clusterRolePredicates = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		notAllowedNames := []string{"kserve-admin"}
```

Comment:
For some reason, this ClusterRoleBinding seems to cause periodic reconciliation. Watching the resource with `watch --differences=permanent oc get clusterrole kserve-admin -o yaml` just shows the resourceVersion changing, but then if I diff the `objectNew` and `objectOld`, I see that the `rules` section has changed (null in `objectNew`), so the `compareHashPredicate` that I've created below isn't enough to avoid it continuously triggering reconciliation:

![image](https://github.com/user-attachments/assets/12c0cfc0-a477-4ed3-9a01-0308e5456a72)

Given that the manifest has an empty rules list, I _guess_ it's as a result of our previous reconcile that it's null in ObjectNew, but then if so, I don't know what's populating the rules. Thoughts?

Reply:
I guess this is because the `kubeflow-kserve-admin` is an aggregation so the actual rules are computed by kubernetes. Probably for this we need an handler that if  `aggregationRule`  is present, then it ignores the `rules` field, maybe something like https://github.com/opendatahub-io/opendatahub-operator/blob/acd89b8846393d17e0715cafce00237961911261/pkg/controller/predicates/dependent/dependent.go#L78-L114

Reply:
I've changed it now to be a `clusterRoleAggregationRulePredicate`. I guess neither that nor the `compareHashPredicate` here are specific to the kserve component now...should I pull those out to `pkg/controller/predicates/` somehow/somewhere now, or leave it until they're needed by something else?

Reply:
No objections to move it to the `pkg/controller/predicates` as part of this PR